### PR TITLE
Add automation for version bumps (patch releases only)

### PR DESCRIPTION
**Pull request by bmorelli25**
### Summary

This PR adds a workflow to automate the creation of a version bump PR for Elastic Stack patch releases.

#### Why

Patch releases make up approximately 79% of Elastic Stack releases and occur roughly once every two weeks. For the documentation release manager, the list of tasks includes, but is definitely not limited to:

- open an issue in elastic/dev
- open a PR in elastic/docs
- open a PR in elastic/docs.elastic.co
- full rebuild (going away with BuildKite builds)

I'd like to automate each of these steps. This PR is the first automation attempt. 

#### How does it work

To open a version bump, a user simply needs to input the version that is being released and click "run". Within 30 seconds a PR will open.
Screenshot 2024-01-24 at 3 56 38 PM

#### Notes

This was tested in a private repo and works. We should double-check after this is merged to ensure that Elastic org perms don't break anything.

#### Next steps

* Add similar automation to elastic/docs.elastic.co https://github.com/elastic/docs.elastic.co/pull/200
* Automate issue creation (in progress)
* Single automation to trigger all three (PR, PR, issue) when managing a release (later)
* Add automation for minor releases (later)

### CODE REVIEW

File: .github/workflows/patch-release-version-bump.yml [hunk 1]
```diff
@@ -5,40 +5,15 @@ on:
...
-        - '3'
-        - '4'
-        - '5'
-        - '6'
-        - '18'
-        - '19'
-        - '20'
-        - '21'
-        - '22'
+        options: ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25']
```
Comment (glitteringkatie) [inline .github/workflows/patch-release-version-bump.yml:16]: Let's remove 0 so it isn't confused for a minor bump!

Comment (bmorelli25) [reply .github/workflows/patch-release-version-bump.yml:16]:
```suggestion
        options: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25']
```

Comment (glitteringkatie) [reply .github/workflows/patch-release-version-bump.yml]: Beautiful

**Review by glitteringkatie — APPROVED** [review]
LGTM!

**Pull request merged**